tensorboard/webapp/widgets/histogram/histogram_v2_component.ts

```diff
@@ -0,0 +1,193 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  Input,
+  OnChanges,
+  ViewChild,
+} from '@angular/core';
+
+import * as d3 from '../../third_party/d3';
+import {
+  ColorScale,
+  HistogramData,
+  HistogramMode,
+  TimeProperty,
+} from './histogram_types';
+
+@Component({
+  selector: 'tb-histogram-v2',
+  templateUrl: 'histogram_v2_component.ng.html',
+  styleUrls: ['histogram_v2_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class HistogramV2Component implements OnChanges {
+  @ViewChild('xAxis') private readonly xAxis!: ElementRef;
+  @ViewChild('yAxis') private readonly yAxis!: ElementRef;
+  @ViewChild('content') private readonly content!: ElementRef;
+
+  @Input() mode: HistogramMode = HistogramMode.OFFSET;
+
+  @Input() timeProperty: TimeProperty = TimeProperty.STEP;
+
+  @Input() colorScale?: ColorScale;
+
+  @Input() name!: string;
+
+  @Input() data!: HistogramData;
+
+  private yScale?:
+    | d3.ScaleTime<number, number>
+    | d3.ScaleLinear<number, number>;
+
+  constructor(private readonly host: ElementRef) {}
+
+  ngOnChanges() {
+    if (this.content) {
```

Comment:
Should we be skipping updateChart() when only the colorScale/name changes?

Reply:
That'd be an optimization to be done later.